web/hooks/use-comments.ts

```diff
@@ -20,12 +18,14 @@ export const useComments = (contractId: string) => {
 
   return comments
 }
-export const useCommentsOnPost = (postId: string | undefined) => {
-  const [comments, setComments] = useState<PostComment[] | undefined>()
 
-  useEffect(() => {
-    if (postId) return listenForCommentsOnPost(postId, setComments)
-  }, [postId])
-
-  return comments
+export const useNewCommentsOnPost = (postId: string) => {
+  return useRealtimeRows('post_comments', { k: 'post_id', v: postId }).map(
```

Comment:
useRealtimeRows misses updates when you are off the tab. Better to use `useSubscription` and pass in a fetch function to refetch data while you were away.

Reply:
Note that you need to fetch it the raw rows array. See use-comments-supabase.ts as an example (modified today):

```ts
export function useRealtimeComments(limit: number): Comment[] {
  const { rows } = useSubscription('contract_comments', undefined, () =>
    getCommentRows(limit)
  )
  return (rows ?? []).map((r) => r.data as Comment)
}
```